cmd/check_telemetry_test.go

```diff
@@ -0,0 +1,203 @@
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"testing"
+
+	"github.com/CircleCI-Public/circleci-cli/settings"
+	"github.com/CircleCI-Public/circleci-cli/telemetry"
+	"github.com/spf13/afero"
+	"gotest.tools/v3/assert"
+)
+
+type testTelemetry struct {
+	events []telemetry.Event
+}
+
+func (cli *testTelemetry) Close() error { return nil }
+
+func (cli *testTelemetry) Track(event telemetry.Event) error {
+	cli.events = append(cli.events, event)
+	return nil
+}
+
+func TestAskForTelemetryApproval(t *testing.T) {
+	// Mock HTTP
+	userId := "id"
+	uniqueId := "unique-id"
+	response := fmt.Sprintf(`{"id":"%s","login":"login","name":"name"}`, userId)
+	var handler http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, r.Method, "GET")
+		assert.Equal(t, r.URL.String(), "/me")
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte(response))
+	}
+	server := httptest.NewServer(handler)
```

Comment:
Ugh. I'd rather unit tests don't rely on actually opening a server, but sometimes it can't be avoided. ¯\\\_(ツ)_/¯

Reply:
Completely agree, I considered doing otherwise: I would have liked to instead just inject a fake api client but that would have required a way bigger change. I'm completely open to alternative!

Reply:
Alright, thinking about it I think I'm gonna inject it like the terminalInterface